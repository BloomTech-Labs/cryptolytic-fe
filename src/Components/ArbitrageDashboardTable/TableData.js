import React from "react";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  binanceColor: {
    color: "rgb(78, 185, 255)",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  },
  coinbaseColor: {
    color: "rgb(83, 207, 215)",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  },
  bitfinex: {
    color: "rgb(98, 227, 171)",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  },
  tradingPair: {
    display: "flex",
    alignItems: "center",
    paddingLeft: "23%",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)",

    "@media(max-width: 1600px)": {
      padding: "1.2em"
    }
  },
  data: {
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  }
});

const TableData = ({ item, positive, negative, icon1, icon2 }) => {
  const classes = useStyles();

  if (positive && item >= 0) {
    return <td className={positive}>{item + " %"}</td>;
  } else if (negative && negative) {
    return <td className={negative}>{item + " %"}</td>;
  } else if (item === "Binance") {
    return <td className={classes.binanceColor}>{item}</td>;
  } else if (item === "Coinbase Pro") {
    return <td className={classes.coinbaseColor}>{item}</td>;
  } else if (item === "Bitfinex") {
    return <td className={classes.bitfinex}>{item}</td>;
  } else if (icon1 && icon2) {
    return (
      <td className={classes.tradingPair}>
        <img src={icon1} />
        <img src={icon2} />

        {item}
      </td>
    );
  } else {
    return <td className={classes.data}>{item}</td>;
  }
};

export default TableData;
