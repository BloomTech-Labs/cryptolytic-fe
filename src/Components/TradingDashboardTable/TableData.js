import React from "react";
import { withStyles, makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  binanceColor: {
    color: "rgb(78, 185, 255)",

  },
  coinbaseColor: {
    color: "rgb(83, 207, 215)",
 
  },
  bitfinex: {
    color: "rgb(98, 227, 171)",
  },
  positive: {
    color: "green",
  },
  negative: {
    color: "red"
  },
  tradingPair: {
    display: "flex",
    marginLeft: "33%",

  }
});

const TableData = ({ item, positive, negative, icon1, icon2 }) => {
  const classes = useStyles();

  if (positive && item >= 0) {
    return <td className={positive}>{item + " %"}</td>;
  } else if (negative && negative) {
    return <td className={negative}>{item + " %"}</td>;
  } else if (item === "Binance") {
    return <td className={classes.binanceColor}>{item}</td>;
  } else if (item === "Coinbase Pro") {
    return <td className={classes.coinbaseColor}>{item}</td>;
  } else if (item === "Bitfinex") {
    return <td className={classes.bitfinex}>{item}</td>;
  } else if (icon1 && icon2) {
    return (
      <td className={classes.tradingPair}>
        <img src={icon1} />
        <img src={icon2} />

        {item}
      </td>
    );
  } else {
    return <td>{item}</td>;
  }
};

export default TableData;
