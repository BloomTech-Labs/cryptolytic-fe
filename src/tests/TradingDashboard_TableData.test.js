import React from "react";
import { MemoryRouter } from "react-router";
///
import TableData from "../Components/TradingDashboardTable/TableData";
///
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

test("Trading TableData can render without crashing", () => {
  render(<TableData />);
});

// Verify correct color and/or class for positive percentage
// test("Trading TableData class applies correct className for positive integers", () => {
//   render(<TableData item={{ percentage: 1 }} />);
// });

describe("<TableData /> matches verified SnapShot", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<TableData />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
