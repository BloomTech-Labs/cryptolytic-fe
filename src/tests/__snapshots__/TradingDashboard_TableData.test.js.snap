// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableData /> matches verified SnapShot should match snapshot 1`] = `
<td
  className="makeStyles-data-10"
/>
`;
